### PR TITLE
feat: Add resume/suspend endpoints for CronWorkflows

### PR DESCRIPTION
**Pull request by simster7**
Closes: https://github.com/argoproj/argo/issues/4441

Signed-off-by: Simon Behar <simbeh7@gmail.com>

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: pkg/apiclient/cronworkflow/cron-workflow.proto [hunk 1]
```diff
@@ -71,4 +87,18 @@ service CronWorkflowService {
     rpc DeleteCronWorkflow (DeleteCronWorkflowRequest) returns (CronWorkflowDeletedResponse) {
         option (google.api.http).delete = "/api/v1/cron-workflows/{namespace}/{name}";
     }
-}
\ No newline at end of file
+
+    rpc ResumeCronWorkflow (CronWorkflowResumeRequest) returns (github.com.argoproj.argo.pkg.apis.workflow.v1alpha1.CronWorkflow) {
+        option (google.api.http) = {
+			put: "/api/v1/cron-workflows/{namespace}/{name}/resume"
```
Comment (alexec) [inline pkg/apiclient/cronworkflow/cron-workflow.proto:93]: does ResumeWorkflow use PUT?

File: pkg/apiclient/error-translating-cron-workflow-service-client.go [hunk 2]
```diff
@@ -63,3 +63,19 @@ func (c *errorTranslatingCronWorkflowServiceClient) DeleteCronWorkflow(ctx conte
 	}
 	return workflow, err
 }
+
+func (c *errorTranslatingCronWorkflowServiceClient) ResumeCronWorkflow(ctx context.Context, req *cronworkflowpkg.CronWorkflowResumeRequest, _ ...grpc.CallOption) (*v1alpha1.CronWorkflow, error) {
+	workflow, err := c.delegate.ResumeCronWorkflow(ctx, req)
+	if err != nil {
+		return nil, grpcutil.TranslateError(err)
```
Comment (alexec) [inline pkg/apiclient/error-translating-cron-workflow-service-client.go:70]: what about updating `TranslateError` to return nil if `err` is nil? You could then just do this:

```golang
return workflow, grpcutil.TranslateError(err)
```

File: server/cronworkflow/cron_workflow_server.go [hunk 3]
```diff
@@ -90,6 +90,23 @@ func (c *cronWorkflowServiceServer) DeleteCronWorkflow(ctx context.Context, req
...
+	return c.setCronWorkflowSuspend(true, req.Namespace, req.Name, ctx)
+}
+
+func (c *cronWorkflowServiceServer) setCronWorkflowSuspend(setTo bool, namespace, name string, ctx context.Context) (*v1alpha1.CronWorkflow, error) {
+	cwf, err := c.getCronWorkflowAndValidate(ctx, namespace, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	cwf.Spec.Suspend = setTo
+	return auth.GetWfClient(ctx).ArgoprojV1alpha1().CronWorkflows(namespace).Update(cwf)
```
Comment (alexec) [inline server/cronworkflow/cron_workflow_server.go:107]: should we use patch to avoid conflicts?

File: test/e2e/argo_server_test.go [hunk 4]
```diff
@@ -898,6 +898,24 @@ func (s *ArgoServerSuite) TestCronWorkflowService() {
 			Status(200)
 	})
 
+	s.Run("Suspend", func() {
+		s.e().PUT("/api/v1/cron-workflows/argo/test/suspend").
```
Comment (alexec) [inline test/e2e/argo_server_test.go:902]: nice

File: ui/src/app/shared/services/cron-workflow-service.ts [hunk 5]
```diff
@@ -30,4 +30,12 @@ export class CronWorkflowService {
     public delete(name: string, namespace: string) {
         return requests.delete(`api/v1/cron-workflows/${namespace}/${name}`);
     }
+
+    public suspend(name: string, namespace: string) {
+        return requests.put(`api/v1/cron-workflows/${namespace}/${name}/suspend`).then(res => res.body as CronWorkflow);
```
Comment (alexec) [inline ui/src/app/shared/services/cron-workflow-service.ts:35]: excellent

File: pkg/apiclient/cronworkflow/cron-workflow.proto (hunk 1 above)
Comment (simster7) [reply pkg/apiclient/cronworkflow/cron-workflow.proto:93]: Yes it does

File: pkg/apiclient/error-translating-cron-workflow-service-client.go (hunk 2 above)
Comment (simster7) [reply pkg/apiclient/error-translating-cron-workflow-service-client.go:70]: Separate PR?

File: server/cronworkflow/cron_workflow_server.go (hunk 3 above)
Comment (simster7) [reply server/cronworkflow/cron_workflow_server.go:107]: Sure, but I would like to do the same for Workflow resume/suspend for consistency. Separate PR?

**Comment by alexec** [comment]
Needs code owner approval from @jessesuen

**Review by jessesuen — APPROVED** [review]

File: pkg/apiclient/error-translating-cron-workflow-service-client.go (hunk 2 above)
Comment (simster7) [reply pkg/apiclient/error-translating-cron-workflow-service-client.go:70]: Opened #4465

File: server/cronworkflow/cron_workflow_server.go (hunk 3 above)
Comment (simster7) [reply server/cronworkflow/cron_workflow_server.go:107]: Opened #4468

**Pull request merged**